### PR TITLE
463 profileuserid profile

### PR DESCRIPTION
**Pull request by Sylvester2003**
### Summary
🎖️🌐🛠️

This pull request improves the localization, functionality, and user experience of the frontend application. It updates the `nuxt.config.ts` file and the locale files to use country-specific formats. It adds new features and logic to the user profile page, such as badges, followers, following, and profile picture. It adds new components, such as `Badge.vue` and `Modal.vue`, and modifies existing components, such as `LanguageSwitcher.vue` and `TheNavbar.vue`, to enhance the UI and navigation. It simplifies and fixes some code in the pages and stores, such as `create.vue`, `recipes/[userId].vue`, `recipe.ts`, and `image.ts`. It removes some unnecessary or debugging code, such as console.log statements.

> _New features and fixes_
> _`Badge.vue` and `Modal.vue` shine_
> _Winter code review_

### Walkthrough
*  Add new components for displaying badges, modal windows, and searchable dropdown menus ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-45937ff2d5148dca436efea1ca22bd6ddb27b5448b8fd397fb9431fda948d0b4R1-R21), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-45b0067eca93ce4d067beff9bd7976ecdfb1477376643e99290137ce0b8e8000R1-R45), )
*  Modify existing components to use SearchSelect, localePath, and navigateTo functions, and fix some class names and console.log statements ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-8c067f079be696542b0372cb03ba9e161d337119ef95ff1b9a3f2e78561ede7aL3-R33), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-0eea7036cf3872a5b69426a1b42953545854fe49719ca91943ff35ccef4e3700L7-R7), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-348c1d9512523947f977a1b46308365a8049f2c887297f386dfada3bf02b6ae5L18-L19), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-4100b162e90ddbb8951ec90b956b8d8226eed6d822588d7c633e952fb851badbL57-R57), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-daa8b609bcd6527103f63a1cc4a4829db11380361829a89e999255d35771ef00L37-R37))
*  Modify existing pages to add new features and logic for the profile, recipe, and user pages, and simplify the fetching of user and recipe data ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-323fa6e125ac78216ea50f13cedc453074da24aff8ae106e04abc309478cac3dL40-R44), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-ade85aa68406c565b477a1ae17db8373bfbdacc26ce7185dfe997711e81b213eL8-R13), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-ade85aa68406c565b477a1ae17db8373bfbdacc26ce7185dfe997711e81b213eL21-R26), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-0c5eabb00b3c5ba6eaf821bc06b45da7c540b03457b2d0d259239909abcfafb6L14-R19), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-0c5eabb00b3c5ba6eaf821bc06b45da7c540b03457b2d0d259239909abcfafb6L23-R25), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-0c5eabb00b3c5ba6eaf821bc06b45da7c540b03457b2d0d259239909abcfafb6L33-R40), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-999bb031ef4df98457ba2dff5a2f438d79e4e880b07184ddae5da6df1a51ab55L3-R89), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-0a4d3fe02d20b046fda6d5837ba84eb00c4b566292a1e81692da4c53b3647de4L7-R9), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-9e5b14e3af2fb27ab7ea51fc5af451bae5975a5625394f5361633db38c85cdcdL2-R2), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-9e5b14e3af2fb27ab7ea51fc5af451bae5975a5625394f5361633db38c85cdcdL11-R11), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-85e1e9fc6db577ab51fbfba2bcd9f574a0d3be2823bc3fa604aceec3d9b6b10fL3-R51), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-85e1e9fc6db577ab51fbfba2bcd9f574a0d3be2823bc3fa604aceec3d9b6b10fL21-R57))
*  Modify existing stores to add new state and actions for image, recipe, and user data, and translate the volume of linked recipes ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-1fc7a050bc25532b502b63820b4eb45b35dff2e4e5886aec9294fac555e37be2L42-R46), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-fff9d1299723af9e15255bb8947c21b5d5453a9ef83d63926b63ae8b8f456b2eL116-R120), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-fff9d1299723af9e15255bb8947c21b5d5453a9ef83d63926b63ae8b8f456b2eL180-R181), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-fff9d1299723af9e15255bb8947c21b5d5453a9ef83d63926b63ae8b8f456b2eR213-R230), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-51c2edcfdddf30bd8958554fb52c8d1e3b795743314b1b06a31a48cfe795ace4R35), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-99a55108f8fc7f5654915d714ec99471d2ff4c429feecaec77d02010f186d877R4), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-99a55108f8fc7f5654915d714ec99471d2ff4c429feecaec77d02010f186d877L80-R90), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-99a55108f8fc7f5654915d714ec99471d2ff4c429feecaec77d02010f186d877R122-R185))
*  Modify existing nuxt.config.ts file to add country codes to locale codes and update the defaultLocale option ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-7cc25f413b350c923da4c68e6215dc8a8ffabb8b616c8326715e6899263ad22cL45-R70), [link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-7cc25f413b350c923da4c68e6215dc8a8ffabb8b616c8326715e6899263ad22cL77-R77))
*  Modify existing locale file en_US.json to add a new string for translating the volume of linked recipes ([link](https://github.com/14A-A-Lyedlik-Devs/Gump/pull/582/files?diff=unified&w=0#diff-4082c608a443cbcd30ab810bf0e9ab801b08c94fc8c3017ca4287456f204e961L75-R76))

### CODE REVIEW

File: apps/frontend/Gump/components/recipe/RecipeFooter.vue [hunk 1]
```diff
@@ -54,7 +54,7 @@ async function saveClick() {
 <template>
   <div v-if="props.recipe" flex="~ row" mx-2 justify-between text-left font-mono text-xl>
     <div flex="~ row" items-center>
-      <div i-fa6-solid-eye orangeIcon />
+      <div orangeicon i-fa6-solid-eye />
```
Comment (Rettend) [inline apps/frontend/Gump/components/recipe/RecipeFooter.vue:57]: . ....

File: apps/frontend/Gump/stores/recipe.ts [hunk 2]
```diff
@@ -113,10 +113,11 @@ export const useRecipeStore = defineStore('recipe', {
         this.currentRecipe.ingredients.splice(index, 1)
     },
     addRecipe(recipe: Recipe) {
+      const i18n = useI18n()
       this.currentRecipe?.ingredients.push({
         name: recipe.title,
         value: 1,
-        volume: 'piece',
+        volume: i18n.t('LinkedRecipeVolume'),
```
Comment (Rettend) [inline apps/frontend/Gump/stores/recipe.ts:120]: ilyen van? egyébként a linkedRecipe !== 0 ingredienteknél amúgy se látszik a volume

File: apps/frontend/Gump/stores/user.ts [hunk 3]
```diff
@@ -123,6 +119,70 @@ export const useUserStore = defineStore('user', {
...
+      const { data, error } = await gumpFetch('user/update', {
+        method: 'PATCH',
+        body: JSON.stringify(requestBody),
+      }).json()
+      if (data.value)
+        return data.value
+      if (error.value)
+        return error.value
+    },
+    async getBadgeById(id: number): Promise<Badge | undefined> {
```
Comment (Rettend) [inline apps/frontend/Gump/stores/user.ts:138]: A Badge-nek lehetne egy új store inkább nem a userbe legyen. lehet használni a useStore factory-t

**Review by Rettend — CHANGES_REQUESTED** [review]

File: apps/frontend/Gump/stores/recipe.ts (hunk 2 above)
Comment (Sylvester2003) [reply apps/frontend/Gump/stores/recipe.ts:120]: Amúgy teljesen működőképes ez a megoldás, de ha ha felesleges, akkor kiveszem.

File: apps/frontend/Gump/stores/user.ts (hunk 3 above)
Comment (Sylvester2003) [reply apps/frontend/Gump/stores/user.ts:138]: Oké

File: apps/frontend/Gump/stores/badge.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+export const useBadgeStore = defineStore('badge', {
```
Comment (Rettend) [inline apps/frontend/Gump/stores/badge.ts:1]: use generic store.
Csak 8 fajta badge van és 1 badge az összes usernak "ugyanaz".  Nem kell folyton lekérni mindegyiket egyesével.

File: apps/frontend/Gump/stores/recipe.ts (hunk 2 above)
Comment (Rettend) [reply apps/frontend/Gump/stores/recipe.ts]: mármint értem hogy beraktál egy új key-t az angolba de
1. arra van Localer hogy ne sneakybe editeld itt a json-t >:(
2. nem fog már senki se lefordítani semmit
3. ha van egy linkedRecipe egy ingredientnél akkor csak a name látszik (még a value-t odarakom akkor de a volume szerintem nem kell)

**Pull request merged**